spark/src/main/scala/org/apache/comet/serde/operator/CometDataWritingCommand.scala

```diff
@@ -163,7 +166,29 @@ object CometDataWritingCommand extends CometOperatorSerde[DataWritingCommandExec
         other
     }
 
-    CometNativeWriteExec(nativeOp, childPlan, outputPath)
+    // Create FileCommitProtocol for atomic writes
+    val jobId = java.util.UUID.randomUUID().toString
+    val committer =
+      try {
+        // Use Spark's SQLHadoopMapReduceCommitProtocol
+        val committerClass =
+          classOf[org.apache.spark.sql.execution.datasources.SQLHadoopMapReduceCommitProtocol]
```

Comment:
The commit protocol class is configurable in spark. We should add a fallback when the configuration is not `SQLHadoopMapReduceCommitProtocol`.

https://github.com/apache/spark/blob/4435a3a7f0883559de3bd13ba84f311200943622/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L2194-L2203

Reply:
+1